lib/cpl.rb

```diff
@@ -204,6 +206,10 @@ def show_info_header(config)
       # Add a newline after the info header
       puts
     end
+
+    def first_run?
+      ObjectSpace.each_object(self.class).count == 1
```

Comment:
Is this the best way to do this? In other places, we use instance/class variables:

![omHjTIfJsh](https://github.com/shakacode/heroku-to-control-plane/assets/25509361/a5936796-6bf5-4612-b318-399b12d600b6)

Reply:
I am not happy about it either but this was the only way I could implement it.

I tried this at first. But when we run a command (run CLI start method) in another command, we get into a new instance, and we don't have access to the earlier instance variables.

Reply:
This raises a WTF...

Where in the world did you get the ObjectSpace idea?

Counting object instances is a total hack.

You could probably set a **static** counter on or boolean on the CPL class. In other words, instance variable on the class, not the instance.

Use a class instance variable (aka static) on the cpl class.

https://chat.openai.com/share/e62c8db1-0622-4617-b8a6-5c3a4d82f387